packages/plugin/tests/e2e.rs

```diff
@@ -0,0 +1,75 @@
+use std::sync::Arc;
+
+use polywrap_client::client::PolywrapClient;
+use polywrap_client::core::uri::Uri;
+use polywrap_client_builder::{PolywrapClientConfig, PolywrapClientConfigBuilder};
+use polywrap_core::{error::Error, macros::uri};
+use polywrap_plugin::{error::PluginError, package::PluginPackage};
+use polywrap_tests_utils::mocks::{ArgsSetData, MemoryStoragePlugin};
+
+#[test]
+fn invoke_methods() {
```

Comment:
would you agree to move this test to the client? the main reason would be that this way, we're introducing four new dependencies to the plugin crate (even tho they are dev dependencies); and when it comes to releasing the `plugin` crate is released before the `client` or `client-builder`

Reply:
Not sure. I think it makes more sense here since client doesn't need to know about plugins (or wasm wrappers) only the wrapper trait.
That said, it'd be better if these were unit tests.
Also, since we're in a workspace does it matter we have circular dev-dep?

Reply:
> I think it makes more sense here since client doesn't need to know about plugins (or wasm wrappers) only the wrapper trait.

I agree with you, but in our current structure, the client already depends on `test-utils` crate (https://github.com/polywrap/rust-client/blob/main/packages/client/Cargo.toml#L26), and you don't need to add any new dependency to any crate

> it'd be better if these were unit tests.

i completely agree with you, but for the sake of this PR I think this test (moved to client) is good enough

> Also, since we're in a workspace does it matter we have circular dev-dep?

cargo is very annoying with the dependencies (this is actually not adding a circular dependency but rather just a new dependency to the plugin crate); when it comes to the release process, we are publishing each crate in order, respecting the dependency order of each one https://github.com/polywrap/rust-client/blob/main/.github/scripts/release.sh#L4-L16. if you check the script, you can see that `plugin` is being published **before** `client` and `client builder` crates 
so just by adding these dependencies, the script of release would fail because it would expect `client` to be published before `plugin`

(fun fact: we're also publishing `test-utils` just because it's a dependency of the client)